### PR TITLE
Feature: Add option to exclude consumer from Stream metrics

### PR DESCRIPTION
**Pull request by nantiferov**
Tested it manually, looks like working properly, PTAL.

Probably need to add more tests, can check it later.

Amount of metrics on test redis:
* without streams metrics collect - 260
* with `REDIS_EXPORTER_CHECK_STREAMS=*` - 680
* with this feature enabled in addition to previous - 430

Fixes: #902

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -148,6 +148,7 @@ Prometheus uses file watches and all changes to the json file are applied immedi
...
 | check-single-streams    | REDIS_EXPORTER_CHECK_SINGLE_STREAMS    | Comma separated list of streams to export info about streams, groups and consumers. The streams specified with this flag will be looked up directly without any glob pattern matching.  Use this option if you don't need glob pattern matching;  it is faster than `check-streams`.                                                                                                                                                                                                                                                              |
+| streams-exclude-consumer    | REDIS_EXPORTER_STREAMS_EXCLUDE_CONSUMER    | Don't collect per consumer metrics for streams (decreases amount of metrics and cardinality).                                                                                                                                                                                                                                                                                                                                                                                                                   |
```
Comment (oliver006) [inline README.md:151]: REDIS_EXPORTER_STREAMS_EXCLUDE_CONSUMER -> REDIS_EXPORTER_STREAMS_EXCLUDE_CONSUMER_METRICS ?

File: exporter/exporter.go [hunk 2]
```diff
@@ -55,6 +55,7 @@ type Options struct {
 	CheckSingleKeys           string
 	CheckStreams              string
 	CheckSingleStreams        string
+	StreamsExcludeConsumer    bool
```
Comment (oliver006) [inline exporter/exporter.go:58]: Same, add `Metrics` ?

**Review by oliver006 — COMMENTED** [review]
Thanks for the PR!
Looks good, minor suggestions re: naming to make it more clear what you're excluding (metrics) but I think you're going in the right direction.

File: README.md (hunk 1 above)
Comment (nantiferov) [reply README.md:151]: Yep, sure.

My only concern was that it would be quite long, but not too much I think)

Comment (nantiferov) [reply README.md]: Done, committed changes.

File: exporter/exporter.go (hunk 2 above)
Comment (nantiferov) [reply exporter/exporter.go]: Done

File: README.md (hunk 1 above)
Comment (nantiferov) [reply README.md]: Apparently tests are failing, but I don't see exact reason.

In the end of output only 

```log
FAIL
coverage: 90.0% of statements
FAIL	github.com/oliver006/redis_exporter/exporter	22.273s
FAIL
```

```log

```

**Comment by oliver006** [comment]
Try rebasing your PR, I pushed a fix to master last week to fix an issue with an outdated Redis docker container.

**Comment by nantiferov** [comment]
> Try rebasing your PR, I pushed a fix to master last week to fix an issue with an outdated Redis docker container.

That helped, thank you.

Though coverage decreased, I will check later today to add tests.

File: README.md [hunk 3]
```diff
@@ -148,6 +148,7 @@ Prometheus uses file watches and all changes to the json file are applied immedi
...
 | check-single-streams    | REDIS_EXPORTER_CHECK_SINGLE_STREAMS    | Comma separated list of streams to export info about streams, groups and consumers. The streams specified with this flag will be looked up directly without any glob pattern matching.  Use this option if you don't need glob pattern matching;  it is faster than `check-streams`.                                                                                                                                                                                                                                                              |
+| streams-exclude-consumer    | REDIS_EXPORTER_STREAMS_EXCLUDE_CONSUMER_METRICS    | Don't collect per consumer metrics for streams (decreases amount of metrics and cardinality).                                                                                                                                                                                                                                                                                                                                                                                                                   |
```
[the remark this answers is not in the document]
Comment (nantiferov) [reply README.md]: Thanks for noticing. Missed update here.

Now should be good.

File: exporter/streams_test.go [hunk 4]
```diff
@@ -544,3 +544,59 @@ func TestStreamsExtractStreamMetrics(t *testing.T) {
 
 	}
 }
+
+func TestStreamsExtractStreamMetricsExcludeConsumer(t *testing.T) {
```
Comment (oliver006) [inline exporter/streams_test.go:548]: I think this is missing a check for metrics that are *not* present - as in, metrics that normally would be there but due to the `StreamsExcludeConsumerMetrics` check are now not there. Let's add that.

Comment (nantiferov) [reply exporter/streams_test.go:548]: Yep, I though about this, but haven't added initially)

Now committed change to `TestStreamsExtractStreamMetricsExcludeConsumer` test to check for metrics that shouldn't be collected. PTAL.

P.S. Probably not ideal implementation, but works fine.

**Review by oliver006 — APPROVED** [review]
Looks good now, thanks for the PR!

**Comment by nantiferov** [comment]
@oliver006 when you have time, could you please merge this PR if all looks good now and release a new version?

**Comment by oliver006** [comment]
Thanks for the nudge.
Do you think we can get #911 sorted and into the next release as well?

**Comment by nantiferov** [comment]
Yep, probably can check it later today.

**Comment by oliver006** [comment]
Released [v1.60.0](https://github.com/oliver006/redis_exporter/releases/tag/v1.60.0) which includes this and all your other PRs - nice work, thanks for all your contributions!

**Comment by nantiferov** [comment]
Thank you!

**Pull request merged**